microprofile/fault-tolerance/src/test/java/io/helidon/microprofile/faulttolerance/AsynchronousBean.java

```diff
@@ -165,4 +169,34 @@ CompletableFuture<String> asyncCompletableFutureWithFallbackFailure() {
         future.completeExceptionally(new IOException("oops"));
         return future;
     }
+
+    /**
+     * Normal asynchronous call.
+     *
+     * @return A future.
+     */
+    @Asynchronous
+    @WithExecutor("platform-thread-executor")
+    CompletableFuture<String> asyncWithExecutor() {
+        called.set(!Thread.currentThread().isVirtual());
+        FaultToleranceTest.printStatus("AsynchronousBean::asyncWithExecutor", "success");
+        return CompletableFuture.completedFuture("success");
+    }
+
+    @Produces
+    @WithExecutor("platform-thread-executor")
+    ExecutorService executorService() {
+        return Executors.newFixedThreadPool(10);
+    }
+
+    /**
+     * Bad executor name, should throw a {@link FaultToleranceException}.
+     *
+     * @return A future.
+     */
+    @Asynchronous
+    @WithExecutor("does-not-exist")
```

Comment:
Throwing an exception at runtime is not very user friendly.
Is there a chance to validate this during startup? Or maybe have an option to fallback to the default executor if the named one is not defined?

Reply:
We could validate at startup I suppose. Silently falling back to the default one does not seem like a good idea to me when the task may depend on the right executor being used.